src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -67,8 +67,12 @@ export const RoomEditModal: React.FC<RoomEditModalProps> = ({
   );
 
   const onSubmit = useCallback(() => {
-    return submitHandler(values, room.roomIndex!);
-  }, [room.roomIndex, submitHandler, values]);
+    const roomValues = {
+      ...room,
+      ...values,
+    };
+    return submitHandler(roomValues, room.roomIndex!);
```

Comment:
We want to eliminate this `!` TypeScript operator from our codebase. It disables type checking and says that there can never be an `undefined` value here, which has lead to weird/hard to find bugs:

```suggestion
    return submitHandler(roomValues, room.roomIndex);
```

This is likely going to need some other related checks/type changes as well once this change is made.